client/src/modules/Teams/components/JoinTeamModal/JoinTeamModal.tsx

```diff
@@ -0,0 +1,69 @@
+import { Form, Input, Typography } from 'antd';
+import Modal from 'antd/lib/modal/Modal';
+import { JoinTeamDto } from 'api';
+import { passwordPattern } from 'services/validators';
+
+type Props = {
+  onCancel: () => void;
+  onSubmit: (id: number, record: JoinTeamDto) => Promise<void>;
+};
+
+const { Text } = Typography;
+
+const layout = {
+  labelCol: { span: 6 },
+  wrapperCol: { span: 18 },
+};
+
+export default function JoinTeamModal({ onCancel, onSubmit }: Props) {
+  const [form] = Form.useForm<Partial<JoinTeamDto>>();
+
+  const handleModalSubmit = async (values: Partial<JoinTeamDto>) => {
+    const [id, password] = values.password!.split('_');
+
+    if (!password || !Number(id)) {
+      return;
+    } else {
+      await onSubmit(Number(id), { password: password });
+    }
+  };
```

Comment:
wouldn't it be better to check password existence in values instead of using non-null assertion operator? 

you could use just `{ password }`.

Reply:
![image](https://user-images.githubusercontent.com/57412523/212969897-bd4dd90e-11e1-4f9b-9337-5532037cf930.png)
we have validation, so the password will always be in the right format

{ password } - fixed